lib/util/Components.js

```diff
@@ -110,7 +109,7 @@ class Components {
       // preserving original array so it can be merged later on.
       copyUsedPropTypes = this._list[id].usedPropTypes && this._list[id].usedPropTypes.slice();
     }
-    this._list[id] = util._extend(this._list[id], props);
+    this._list[id] = Object.assign(this._list[id], props);
```

Comment:
this should either be `Object.assign(this._list[id], props);` or `this._list[id] = Object.assign({}, this._list[id], props);` - iow, if it's mutating, it shouldn't reassign.